marimo/_ai/llm/_impl.py

```diff
@@ -664,3 +702,601 @@ def __call__(
+            Model object, or a pre-configured pydantic_ai.Agent instance.
+            When passing an Agent, other parameters are ignored.
+        tools: List of tool functions for the model to use (ignored if Agent).
+        instructions: System instructions for the model (ignored if Agent).
+        model_settings: Model-specific settings (e.g., AnthropicModelSettings
+            for thinking). Can be used with both model strings and Agents.
+        **kwargs: Additional arguments passed to pydantic_ai.Agent constructor.
+            See https://ai.pydantic.dev/agents/ for all options.
+
+    Example - Simple usage (like other mo.ai.llm classes):
+        ```python
+        chat = mo.ui.chat(
+            mo.ai.llm.pydantic_ai(
+                "openai:gpt-4.1",
+                tools=[get_weather, calculate],
+                instructions="You are a helpful assistant.",
+            )
+        )
+        ```
+
+    Example - With thinking enabled:
+        ```python
+        from pydantic_ai.models.anthropic import AnthropicModelSettings
+
+        chat = mo.ui.chat(
+            mo.ai.llm.pydantic_ai(
+                "anthropic:claude-sonnet-4-5",
+                tools=[get_weather],
+                instructions="Think step by step.",
+                model_settings=AnthropicModelSettings(
+                    max_tokens=8000,
+                    anthropic_thinking={
+                        "type": "enabled",
+                        "budget_tokens": 4000,
+                    },
+                ),
+            )
+        )
+        ```
+
+    Example - Using a pre-configured Agent (full control):
+        ```python
+        from pydantic_ai import Agent
+
+        # Create and fully configure your own Agent
+        agent = Agent(
+            "anthropic:claude-sonnet-4-5",
+            tools=[get_weather],
+            deps_type=MyDeps,
+            output_type=MyOutput,
+            # ... any other Agent options
+        )
+
+        chat = mo.ui.chat(mo.ai.llm.pydantic_ai(agent))
+        ```
+
+    Example - W&B Inference with custom model:
+        ```python
+        from pydantic_ai.models.openai import OpenAIChatModel
+        from pydantic_ai.providers.openai import OpenAIProvider
+        from pydantic_ai.profiles.openai import OpenAIModelProfile
+
+        model = OpenAIChatModel(
+            model_name="deepseek-ai/DeepSeek-R1-0528",
+            provider=OpenAIProvider(
+                api_key="your-wandb-key",
+                base_url="https://api.inference.wandb.ai/v1",
+            ),
+            profile=OpenAIModelProfile(
+                openai_chat_thinking_field="reasoning_content",
+            ),
+        )
+
+        chat = mo.ui.chat(
+            mo.ai.llm.pydantic_ai(model, instructions="Think step by step.")
+        )
+        ```
+    """
+
+    def __init__(
+        self,
+        model: Any,
+        *,
+        tools: Optional[list[Callable[..., Any]]] = None,
```

Comment:
having model and tools separate is still not ideal because that implies all providers/models can use tools (which is not true). allowing just an Agent object helps prevent this